edx_when/api.py

```diff
@@ -247,13 +261,17 @@ def set_date_for_block(course_id, block_id, field, date_or_timedelta, user=None,
 
     with transaction.atomic():
         try:
-            existing_date = models.ContentDate.objects.get(course_id=course_id, location=block_id, field=field)
+            existing_date = models.ContentDate.objects.select_related('policy').get(
+                course_id=course_id, location=block_id, field=field
+            )
+            needs_save = not existing_date.active
             existing_date.active = True
         except models.ContentDate.DoesNotExist:
             if user:
                 raise MissingDateError(block_id)
             existing_date = models.ContentDate(course_id=course_id, location=block_id, field=field)
             existing_date.policy, __ = models.DatePolicy.objects.get_or_create(**date_kwargs)
+            needs_save = True
```

Comment:
Might be worth checking whether the policy is actually different before updating/saving it.

Basically, do the get-or-create first, check if it's newly created, check if it's different than `existing_date.policy`, and then modify `existing_date.policy`.

Reply:
For this block, I don't think that would make sense. This is the "content date doesn't exist, so we have to make it" code path. So we need to set policy and save regardless.

Reply:
ah, got it